From 5c4de60bdb2b267ef97880cd70ed6116f8c15f8e Mon Sep 17 00:00:00 2001
From: Thejas Nair <thejas@hortonworks.com>
Date: Tue, 30 Jun 2015 20:06:24 -0700
Subject: [PATCH 0986/1164] HIVE-11147 : MetaTool doesn't update FS root
 location for partitions with space in name (Wei
 Zheng via Thejas Nair)

Change-Id: I5206d6b810234c40cde499866bbf2d2d7d46a163
---
 .../apache/hadoop/hive/metastore/ObjectStore.java  |   26 +++++++-------------
 1 file changed, 9 insertions(+), 17 deletions(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
index 76171a5..2e36f42 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/ObjectStore.java
@@ -23,7 +23,6 @@
 import java.io.IOException;
 import java.net.InetAddress;
 import java.net.URI;
-import java.net.URISyntaxException;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
@@ -63,6 +62,7 @@
 import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.conf.Configurable;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.hive.common.FileUtils;
 import org.apache.hadoop.hive.common.ObjectPair;
 import org.apache.hadoop.hive.common.classification.InterfaceAudience;
@@ -6069,10 +6069,8 @@ public UpdateMDatabaseURIRetVal updateMDatabaseURI(URI oldLoc, URI newLoc, boole
         URI locationURI = null;
         String location = mDB.getLocationUri();
         try {
-          locationURI = new URI(location);
-        } catch(URISyntaxException e) {
-          badRecords.add(location);
-        } catch (NullPointerException e) {
+          locationURI = new Path(location).toUri();
+        } catch (IllegalArgumentException e) {
           badRecords.add(location);
         }
         if (locationURI == null) {
@@ -6135,10 +6133,8 @@ private void updatePropURIHelper(URI oldLoc, URI newLoc, String tblPropKey, bool
     if (parameters.containsKey(tblPropKey)) {
       String tablePropLocation = parameters.get(tblPropKey);
       try {
-        tablePropLocationURI = new URI(tablePropLocation);
-      } catch (URISyntaxException e) {
-        badRecords.add(tablePropLocation);
-      } catch (NullPointerException e) {
+        tablePropLocationURI = new Path(tablePropLocation).toUri();
+      } catch (IllegalArgumentException e) {
         badRecords.add(tablePropLocation);
       }
       // if tablePropKey that was passed in lead to a valid URI resolution, update it if
@@ -6286,10 +6282,8 @@ public UpdateMStorageDescriptorTblURIRetVal updateMStorageDescriptorTblURI(URI o
         URI locationURI = null;
         String location = mSDS.getLocation();
         try {
-          locationURI = new URI(location);
-        } catch (URISyntaxException e) {
-          badRecords.add(location);
-        } catch (NullPointerException e) {
+          locationURI = new Path(location).toUri();
+        } catch (IllegalArgumentException e) {
           badRecords.add(location);
         }
         if (locationURI == null) {
@@ -6369,10 +6363,8 @@ public UpdateSerdeURIRetVal updateSerdeURI(URI oldLoc, URI newLoc, String serdeP
           String schemaLoc = mSerde.getParameters().get(serdeProp);
           URI schemaLocURI = null;
           try {
-            schemaLocURI = new URI(schemaLoc);
-          } catch (URISyntaxException e) {
-            badRecords.add(schemaLoc);
-          } catch (NullPointerException e) {
+            schemaLocURI = new Path(schemaLoc).toUri();
+          } catch (IllegalArgumentException e) {
             badRecords.add(schemaLoc);
           }
           if (schemaLocURI == null) {
-- 
1.7.9.5

